### PR TITLE
test: add unit and integration tests for stdlib components (#817)

### LINKED ISSUE #817: test: unit tests for stdlib components (react, unit_test_eval)
Add unit tests for stdlib components and frameworks with zero coverage.

## Scope

| File | Stmts | What's missing | Est. tests |
|------|-------|---------------|------------|
| \`stdlib/components/unit_test_eval.py\` | 71 | JSON loading, Pydantic schema validation, \`set_judge_context()\` formatting — all pure logic, used by the eval CLI | ~10 |
| \`stdlib/components/react.py\` | 33 | \`ReactInitiator\` (goal + tool list formatting, finalizer injection), \`ReactThought\` — pure component logic | ~5 |
| \`stdlib/frameworks/react.py\` | 39 | Loop termination on \`final_answer\`, budget exhaustion, tool response collection — testable with mocked backend | ~6 |

## Notes

- \`unit_test_eval.py\` is the easiest win — pure Pydantic + JSON, no LLM needed, used in production by \`cli/eval/runner.py\`
- The react components are pure formatting logic, straightforward to test
- \`frameworks/react.py\` needs a mocked \`Backend.aact()\` but the core loop logic (termination, budget, tool dispatch) is well-defined
- Part of #813

~21 tests across the three files.

### PR DESCRIPTION
**Pull request by planetf1**
# test: add unit and integration tests for stdlib components (#817)

## Type of PR

- [x] Bug Fix
- [ ] New Feature
- [ ] Documentation
- [x] Other

## Description
- [x] Link to Issue: Fixes #817

Adds 32 tests across 3 files for `stdlib/components/react`, `stdlib/frameworks/react`, and `stdlib/components/unit_test_eval`. These are part of the broader #726 testing epic to boost unit and integration coverage for pure logic and reduce reliance on e2e tests.

### Bug fix: `from_json_file` list misalignment

Review feedback identified a bug in `TestBasedEval.from_json_file()` where examples without user messages caused `inputs`, `targets`, and `input_ids` to fall out of sync. The eval runner (`cli/eval/runner.py`) does positional lookups across these lists, so misalignment would silently score predictions against the wrong reference targets. Fixed with an early `continue` to skip examples that have no user messages, with a new test to cover the scenario.

### Coverage improvement

All three modules went from **0% to near-full coverage** without any e2e infrastructure:

| Module | Before | After |
|--------|--------|-------|
| `mellea/stdlib/frameworks/react.py` | 0% | **100%** |
| `mellea/stdlib/components/react.py` | 0% | **97%** (1 uncovered line: warning log path) |
| `mellea/stdlib/components/unit_test_eval.py` | 0% | **98%** (2 uncovered lines) |

### Approach: ScriptedBackend for orchestrator testing

The `react()` framework orchestrator is tricky to test — it coordinates a loop of `aact()` → `_call_tools()` → check `final_answer`. Previously the only options were mock-heavy unit tests (patching `mfuncs.aact` + `mfuncs._call_tools`, which mirrors internal call structure and breaks on any refactor) or full e2e tests against a real LLM backend (slow, non-deterministic, needs Ollama/API keys).

Instead, these tests use a lightweight `ScriptedBackend` fake that subclasses `Backend` and overrides `_generate_from_context()` to return scripted responses. The key insight is that **real `aact()`, real `_call_tools()`, real `MelleaTool` functions all run** — only LLM inference is faked. This means the tests exercise the actual pipeline (context management, plugin hooks, tool execution) and are robust to internal refactors of `react()`.

These are marked `@pytest.mark.integration` (not unit) because they wire multiple real components together and mock only at the outermost boundary.

### Test breakdown

| File | Tests | Level | What's covered |
|------|-------|-------|----------------|
| `test/stdlib/components/test_react.py` | 10 | Unit | `ReactInitiator` format/parts/parse, `ReactThought` format/parse, finalizer override edge case |
| `test/stdlib/components/test_unit_test_eval.py` | 16 | Unit | Pydantic validation, `format_for_llm`, `set_judge_context` branching (0/1/N targets), `from_json_file` with edge cases (multi-user messages, mixed user messages, non-assistant filtering) |
| `test/stdlib/frameworks/test_react.py` | 7 | Integration | Loop termination, budget exhaustion, non-final tool continuation, model_options tool merge, format branch, multi-tool rejection, context type guard |

### Testing
- [x] Tests added to the respective file if code was changed
- [x] New code has 100% coverage if code as added
- [x] Ensure existing tests and github automation passes (a maintainer will kick off the github automation when the rest of the PR is populated)

### CODE REVIEW

File: test/stdlib/components/test_unit_test_eval.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+            {
+                "input": [{"role": "system", "content": "sys"}],
+                "targets": [],
+                "input_id": "",
+            }
+        ]
+    )
+    path = _write_test_json(tmp_path, data)
+    evals = TestBasedEval.from_json_file(path)
+    assert evals[0].inputs == []
```
Comment (ajbozarth) [inline test/stdlib/components/test_unit_test_eval.py:187]: Comment from Claude:

---

This assertion is correct but incomplete. When an example has no user messages, `targets.append` and `input_ids.append` still run unconditionally, so the lists end up misaligned (`inputs=[], targets=[[]], input_ids=[""]`). Worth either adding the missing assertions to document this as intentional:

    assert evals[0].targets == [[]]
    assert evals[0].input_ids == [""]

...or treating it as a source bug in `from_json_file` — the fix would be `continue` after the `if user_messages:` block so examples with no user message are skipped entirely, keeping all three lists in sync.

---

File: test/stdlib/frameworks/test_react_framework.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            backend=backend,
+            tools=[search],
+            loop_budget=5,
+        )
+
+
+@pytest.mark.asyncio
+async def test_react_rejects_non_chat_context():
+    """react() requires a ChatContext instance."""
+    from unittest.mock import Mock
```
Comment (ajbozarth) [inline test/stdlib/frameworks/test_react_framework.py]: nit: this should be at the top of the file

**Review by ajbozarth — COMMENTED** [review]
I didn't have bandwidth to do a deep review, but I did run the tests and have Claude review, which gave a couple inline comments:

ran `uv run pytest test/stdlib/components/ test/stdlib/frameworks/`:
```console
100 passed, 1 skipped, 1 xpassed, 15 warnings in 159.23s (0:02:39)
```
All new tests passed

File: test/stdlib/components/test_unit_test_eval.py (hunk 1 above)
Comment (planetf1) [reply test/stdlib/components/test_unit_test_eval.py:187]: source bug, not just a test gap. `targets.append` and `input_ids.append` ran unconditionally, so examples without user messages pushed the three lists out of alignment. The eval runner does positional lookups (`targets[idx]`), so it would silently score against the wrong targets.

Fixed in `from_json_file` with an early `continue` when there are no user messages. Updated the existing test to assert all three lists are empty, and added a mixed-example test that would have caught the misalignment.

File: test/stdlib/frameworks/test_react_framework.py (hunk 2 above)
Comment (planetf1) [reply test/stdlib/frameworks/test_react_framework.py]: Moved to module-level imports.

File: test/stdlib/components/test_unit_test_eval.py (hunk 1 above)
Comment (planetf1) [reply test/stdlib/components/test_unit_test_eval.py:187]: decided to include in this issue as the impact is low (vs raising new issue) and just relates to skipping of incomplete examples

**Review by ajbozarth — APPROVED** [review]
Changes LGTM and Claude approve so I'm going to give an approve on this since in general more tests are better than none,
